src/coreclr/vm/dllimportcallback.cpp

```diff
@@ -644,6 +644,9 @@ bool TryGetCallingConventionFromUnmanagedCallersOnly(MethodDesc* pMD, CorInfoCal
         // Set WinAPI as the default
         callConvLocal = MetaSig::GetDefaultUnmanagedCallingConvention();
 
+        bool foundBaseCallConv = false;
+        bool useMemberFunctionVariant = false;
+
         CaValue* arrayOfTypes = &namedArgs[0].val;
         for (ULONG i = 0; i < arrayOfTypes->arr.length; i++)
```

Comment:
>for (ULONG i = 0; i < arrayOfTypes->arr.length; i++) [](start = 8, length = 52)

This loop now seems rather busy. I am curious if it would instead be better to loop through all of them, record what is present using an array of options of `int`s and then loop again for violations and member functions updates.

Reply:
Yeah I'm not a big fan of how complex this loop is and how we basically have to implement it twice. I'll see if I can combine the two implementations and simplify it.